actors/miner/tests/precommit_deposit_and_initial_pledge_positive_test.rs

```diff
@@ -38,10 +38,68 @@ fn initial_pledge_clamped_at_one_attofil() {
         &reward_estimate(),
         &power_estimate(),
         &circulating_supply(),
+        // NOTE: setting this to zero preserves the original pledge definition (before baseline bug fix)
+        //       so these inputs configure the function to return the original pledge
+        0,
+        0,
     );
     assert_eq!(TokenAmount::from_atto(1), initial_pledge);
 }
 
+// Pre-ramp where 'baseline power' dominates
+#[test]
+fn initial_pledge_pre_ramp() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &TokenAmount::from_whole(1),
```

Comment:
I found these a bit hard to mentally verify. Could you please expand them, e.g. by using named vals for the parameters, and spell out in words the expected result. I suggest not using the consts defined for the "clamping" test – mixing and matching those with inline values couples the tests together and makes it hard to trace what the values are intended for. You can probably combine all these new cases into a single test for re-use.

E.g. "With a fixed reward of zero and circulating supply of 1, a sector contributing 1/2 the current network QAP requires pledge of 1/2 * 30% of circulating supply => 15% of one token." 

Please add tests for:
- Another point part-way through the ramp, to demonstrate it's really linear
- Before the ramp start, i.e. epochs_since_ramp_start is negative
- Ramp start and duration of zero (i.e. no ramp for all new networks)
- Ramp duration of zero, and the epochs -1, 0, and 1 around that ramp (i.e. step change)

Reply:
I only have a rudimentary understanding of the reasons behind pledges and FIP0081. My understanding is limited to this: We previously used equation A to calculate initial pledges, and now we're using 70% A + 30% B. If you want a more detailed explanation of the rules, we'll have to pull in one of the authors of FIP0081.


>     Another point part-way through the ramp, to demonstrate it's really linear

Done.

>     Before the ramp start, i.e. epochs_since_ramp_start is negative

Done.

>     Ramp start and duration of zero (i.e. no ramp for all new networks)

Already there.

>     Ramp duration of zero, and the epochs -1, 0, and 1 around that ramp (i.e. step change)

Done.

Reply:
> Ramp start and duration of zero (i.e. no ramp for all new networks)

The existing test is not exercising the pledge calculation properly, it's just testing a boundary condition where there is zero reward and supply. Please add a test for (ramp, duration) = (0, 0) and with reward/supply inputs similar to the other tests you added.

> Ramp duration of zero, and the epochs -1, 0, and 1 around that ramp (i.e. step change)

I don't see this test added. To be concrete, I'm looking to exercise where epochs_since_ramp start differs by zero and +/- 1 from the ramp duration.

Reply:
> > Ramp start and duration of zero (i.e. no ramp for all new networks)
> 
> The existing test is not exercising the pledge calculation properly, it's just testing a boundary condition where there is zero reward and supply. Please add a test for (ramp, duration) = (0, 0) and with reward/supply inputs similar to the other tests you added.

Done.

> > Ramp duration of zero, and the epochs -1, 0, and 1 around that ramp (i.e. step change)
> 
> I don't see this test added. To be concrete, I'm looking to exercise where epochs_since_ramp start differs by zero and +/- 1 from the ramp duration.

Added a test with `epochs_since_ramp={-1,0,1}`.